osu.Game/Screens/Tournament/Drawings.cs

```diff
@@ -0,0 +1,367 @@
+                                            RelativeSizeAxes = Axes.X,
+
+                                            Text = "Reload",
+                                            Action = reloadTeams
+                                        }
+                                    }
+                                },
+                                new FillFlowContainer()
+                                {
+                                    Anchor = Anchor.BottomCentre,
+                                    Origin = Anchor.BottomCentre,
+
+                                    RelativeSizeAxes = Axes.X,
+                                    AutoSizeAxes = Axes.Y,
+                                    Width = 0.75f,
+
+                                    Position = new Vector2(0, -5f),
+
+                                    Direction = FillDirection.Down,
+                                    Spacing = new Vector2(0, 5f),
+
+                                    Children = new Drawable[]
+                                    {
+                                        new OsuButton()
+                                        {
+                                            RelativeSizeAxes = Axes.X,
+
+                                            Text = "Reset",
+                                            Action = () => reset(false)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            float offset = 0;
+            for (int i = 0; i < 6; i++)
+            {
+                visualiserContainer.Add(new VisualiserLine(2 * (float)Math.PI, offset, RNG.Next(10000, 12000))
+                {
+                    RelativeSizeAxes = Axes.Both,
+                });
+
+                offset += (float)Math.PI / 6f;
+            }
+
+            teamsContainer.OnSelected += t =>
+            {
+                groupsContainer.AddTeam(t.Team);
+
+                st.Text = t.Team.FullName;
+                st.FadeIn(200);
+
+                writeResults(groupsContainer.ToStringRepresentation());
+            };
+
+            teamsContainer.OnScrollStarted += () => st.FadeOut(200);
+
+            reset(true);
+        }
+
+        private void writeResults(string text)
+        {
+            lastWriteOp = writeOp;
+            writeOp = Task.Run(async () =>
+            {
+                if (lastWriteOp != null)
+                    await lastWriteOp;
+
+                // Write to drawings_results
+                try
+                {
+                    using (Stream stream = storage.GetStream(results_filename, FileAccess.Write, FileMode.Create))
+                    using (StreamWriter sw = new StreamWriter(stream))
+                    {
+                        sw.Write(text);
+                    }
+                }
+                catch

```

Comment:
In-fact this should not exist at all tbh. Previous results would be stored anyway (so at most it would be offset-by-one) and the way this is implemented/used is thread safe.